core: simplify the streaming subsystem by relying on the common `query()` codepath

Ereski opened the pull request:
This also removes the `before` key on stream events.

Change-type: major
Signed-off-by: Carol Schulze <carol@balena.io>

Ereski committed:
core: simplify the streaming subsystem by relying on the common `query()` codepath

This also removes the `before` key on `data` events, and adds support
for querying through the stream object.

Change-type: major
Signed-off-by: Carol Schulze <carol@balena.io>

grahammcculloch reviewed (commented):
Is there any way to break this PR into smaller chunks? Or are all these UI changes needed in the same release as the backend changes? This is just a terrifyingly large PR to digest :-)

ljewalsh committed:
Update jellyfish-ui-components and jellyfish-chat-widget to versions with updated stream logic

ljewalsh committed:
Updates SupportThread, HomeChannel, Repository lens and View to use the new actions

ljewalsh committed:
Replace loadChannelData, streamViews, and loadViewResults actions with

four new actions: loadChannelData, loadMoreChannelData, loadViewData,
and loadMoreViewData

These actions use the new streaming logic to set up initial streams for
the channels and views. The load more actions can then be used to
paginate

ljewalsh committed:
Ensure stream is stored by the card id even if a slug is passed as the target to loadChannelData or loadMoreChannelData

ljewalsh committed:
Fix inbox pagination

grahammcculloch committed:
Fix Inbox logic

Change-type: patch
Signed-off-by: Graham McCulloch <graham@balena.io>

ljewalsh committed:
Lower number of times test runs

karaxuna committed:
Fix tests, attempt 1

karaxuna committed:
Fix tests, typo

karaxuna committed:
Fix tests, attempt 2

ljewalsh committed:
Update chat-widget to latest version (aligns with latest version of the ui-component package)
